app/routers/temperature.py

```diff
@@ -0,0 +1,66 @@
+import os
+from typing import Optional, List
+import httpx
+from datetime import datetime
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+from .. import crud, schemas, database, models
+from dotenv import load_dotenv
+
+load_dotenv()
+api_key = os.getenv("WEATHER_API_KEY")
```

Comment:
Ensure that the `WEATHER_API_KEY` environment variable is set correctly. If it's missing, the API requests will fail. Consider adding a check to raise an error if the API key is not found.